internal/services/containers/kubernetes_fleet_update_strategy_resource.go

```diff
@@ -0,0 +1,307 @@
+package containers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerservice/2023-10-15/fleetupdatestrategies"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.Resource = KubernetesFleetUpdateStrategyResource{}
+var _ sdk.ResourceWithUpdate = KubernetesFleetUpdateStrategyResource{}
+
+type KubernetesFleetUpdateStrategyResource struct{}
+
+func (r KubernetesFleetUpdateStrategyResource) ModelObject() interface{} {
+	return &KubernetesFleetUpdateStrategyResourceSchema{}
+}
+
+type KubernetesFleetUpdateStrategyResourceSchema struct {
+	FleetId  string                                                         `tfschema:"fleet_manager_id"`
+	Name     string                                                         `tfschema:"name"`
+	Strategy []KubernetesFleetUpdateStrategyResourceUpdateRunStrategySchema `tfschema:"strategy"`
+}
+
+type KubernetesFleetUpdateStrategyResourceUpdateGroupSchema struct {
+	Name string `tfschema:"name"`
+}
+
+type KubernetesFleetUpdateStrategyResourceUpdateRunStrategySchema struct {
+	Stage []KubernetesFleetUpdateStrategyResourceUpdateStageSchema `tfschema:"stage"`
+}
+
+type KubernetesFleetUpdateStrategyResourceUpdateStageSchema struct {
+	AfterStageWaitInSeconds int64                                                    `tfschema:"after_stage_wait_in_seconds"`
+	Group                   []KubernetesFleetUpdateStrategyResourceUpdateGroupSchema `tfschema:"group"`
+	Name                    string                                                   `tfschema:"name"`
+}
+
+func (r KubernetesFleetUpdateStrategyResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return fleetupdatestrategies.ValidateUpdateStrategyID
+}
+
+func (r KubernetesFleetUpdateStrategyResource) ResourceType() string {
+	return "azurerm_kubernetes_fleet_update_strategy"
+}
+
+func (r KubernetesFleetUpdateStrategyResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			ForceNew: true,
+			Required: true,
+			Type:     pluginsdk.TypeString,
+		},
+
+		"fleet_manager_id": commonschema.ResourceIDReferenceRequiredForceNew(fleetupdatestrategies.FleetId{}),
+
+		"strategy": {
+			Required: true,
+			MaxItems: 1,
+			Type:     pluginsdk.TypeList,
+			Elem: &pluginsdk.Resource{
```

Comment:
Since we already know this resource is managing a `strategy` and this can have at most one item we should remove this and just expose the `stage`
```suggestion
```